honeybee_energy/schedule/fixedinterval.py

```diff
@@ -0,0 +1,690 @@
+
+        # set all of the properties that impact how many values can be assigned
+        self._timestep = int_in_range(timestep, 1, 60, 'schedule timestep')
+        assert self._timestep in self.VALIDTIMESTEPS, 'ScheduleFixedInterval timestep ' \
+            '"{}" is invalid. Must be one of the following:\n{}'.format(
+                timestep, self.VALIDTIMESTEPS)
+        assert isinstance(start_date, Date), 'Expected ladybug Date for ' \
+            'ScheduleFixedInterval start_date. Got {}.'.format(type(start_date))
+        self._start_date = start_date
+
+        # set the values and all properties that can be re-set
+        self.name = name
+        self.values = values
+        self.schedule_type_limit = schedule_type_limit
+        self.out_of_range_value = out_of_range_value
+        self.interpolate = interpolate
+
+    @property
+    def name(self):
+        """Get or set the text string for schedule name."""
+        return self._name
+
+    @name.setter
+    def name(self, name):
+        self._name = valid_ep_string(name, 'schedule fixed interval name')
+
+    @property
+    def values(self):
+        """Get or set the schedule's numerical values, which occur at a fixed interval.
+        """
+        return self._values
+
+    @values.setter
+    def values(self, values):
+        self._values = self._check_values(values)
+
+    @property
+    def schedule_type_limit(self):
+        """Get or set a ScheduleTypeLimit object used to assign units to schedule values.
+        """
+        return self._schedule_type_limit
+
+    @schedule_type_limit.setter
+    def schedule_type_limit(self, schedule_type):
+        if schedule_type is not None:
+            assert isinstance(schedule_type, ScheduleTypeLimit), 'Expected ' \
+                'ScheduleTypeLimit for ScheduleRuleset schedule_type_limit. ' \
+                'Got {}.'.format(type(schedule_type))
+        self._schedule_type_limit = schedule_type
+
+    @property
+    def out_of_range_value(self):
+        """Get or set the value to be used for all times not covered by the input values.
+        """
+        return self._out_of_range_value
+
+    @out_of_range_value.setter
+    def out_of_range_value(self, value):
+        self._out_of_range_value = float_in_range(
+            value, input_name='schedule fixed interval out_of_range_value')
+
+    @property
+    def interpolate(self):
+        """Get or set a boolean noting whether values should be interpolated."""
+        return self._interpolate
+
+    @interpolate.setter
+    def interpolate(self, interpolate):
+        self._interpolate = bool(interpolate)
+
+    @property
+    def timestep(self):
+        """Get the integer for the schedule's number of steps per hour."""
+        return self._timestep
+
+    @property
+    def start_date(self):
+        """Get the ladybug Date object noting when the the input values take effect."""
+        return self._start_date
+
+    @property
+    def end_date_time(self):
```

Comment:
I'm sure you have already thought this through but to make sure I understand why this is set-up the way that it is. Does this mean that we can support cases from 21 December to 21 March next year? I _feel_ the end date should be an input by user and we validate start, end and values. This gives a better flexibility but it can also open up cases the the user inputs the wrong values and gets an end date which is not expected without noticing.

Reply:
Yes, I have made sure that everything works right now for schedules that used "reversed analysis_periods" as we call them in ladybug. Adding an `end_date_time` input seems unnecessarily annoying, especially when someone just wants to create a schedule for a single-day simulation and the default value is likely to be wrong. If you want, I could add a method that validates that the `end_date_time` is what someone expects but it will literally be `return self.end_date_time == test_end_date_time` so I'm not sure it's really necessary. Maybe I don't understand what you mean by it adding more "flexibility" since it only seems like it will be more limiting and create more work for the user as I see it now.

Reply:
By "This gives a better flexibility" I meant your current solution gives more flexibility but yes it can stay as is.